joomla/joomla-cms
# Render the debug console with layouts.

Pull Request for Issue # .
Nobody asked for this.

### Summary of Changes
* Render the debug console using layouts
* Write sql log to a `.php` file, not a `.sql` file
* Break up all log files by option/view/etc. 

### Testing Instructions
Turn on debugging. Expect the debug console to work the same as it does without the patch.
Turn on logging (in the plugin settings). 'Deprecated' and 'Everything' logs should now be stored in different files for each site/option/view/layout. Sql Query log files should end in `.php` because (without some specific server settings) `.sql` files could be shown to anyone who knows how to find them and may contain sensitive information that should not be made public.  

### Documentation Changes Required
Probably not?

### Why?
I realize this is kind of a strange PR. Maybe nobody thinks there's much use in using layouts for this when the plugin alone is already working just fine. There are a few good reasons maybe. For one, it greatly reduces the amount of code in the main plugin file making what remains a lot easier to comprehend. Also, it has the benefit that layouts always have which is that they can be easily overridden. Maybe overriding the debug console is not a big priority for most people but I think there can be some good reasons to do so. 

I also think there are several things about this plugin worth examining in more detail. One is the way sql logs are written and overwritten. Another is the question of whether it's appropriate for this plugin to be configure log files at all. 

Anyway, I don't expect this to be merged right away but let's discuss it a bit and see what else can be improved.
